prow/plugins/label/label.go

```
@@ -0,0 +1,116 @@
+	sigMatcher       = regexp.MustCompile(`(?m)@sig-([\w-]*)-(?:misc|test-failures|bugs|feature-requests|proposals|pr-reviews|api-reviews)`)
+	nonExistentLabel = "`%v` does not exist in this repository"
+	labelFailed      = "The following labels: %v could not be added to the issue or PR"
+)
+
+func init() {
+	plugins.RegisterIssueCommentHandler(pluginName, handleIssueComment)
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	IsMember(org, user string) (bool, error)
+	AddLabel(owner, repo string, number int, label string) error
+	GetLabels(owner, repo string) ([]string, error)
+}
+
+func handleIssueComment(pc plugins.PluginClient, ic github.IssueCommentEvent) error {
+	return handle(pc.GitHubClient, ic)
+}
+
+func handle(gc githubClient, ic github.IssueCommentEvent) error {
+	commenter := ic.Comment.User.Login
+	owner := ic.Repo.Owner.Name
+	repo := ic.Repo.Name
+	number := ic.Issue.Number
+
+	labelMatches := labelRegex.FindAllStringSubmatch(ic.Comment.Body, -1)
+	sigMatches := sigMatcher.FindAllStringSubmatch(ic.Comment.Body, -1)
+	if len(labelMatches) == 0 && len(sigMatches) == 0 {
+		return nil
+	}
+
+	member, err := gc.IsMember(repo, commenter)
+	if err != nil {
+		return fmt.Errorf("IsMember failed: %v", err)
+	}
+	if !member {
+		return gc.CreateComment(owner, repo, number, plugins.FormatResponse(ic.Comment, "Only kubernetes org members can add labels."))
+	}
+
+	labels, err := gc.GetLabels(owner, repo)
+	if err != nil {
+		return err
+	}
+	existingLabels := sets.NewString(labels...)
+	var failures []string
+	for _, match := range labelMatches {
+		for _, newLabel := range strings.Split(match[0], " ")[1:] {
+			newLabel = match[1] + "/" + strings.TrimSpace(newLabel)
+			if ic.Issue.HasLabel(newLabel) {
+				continue
+			}
+			if !existingLabels.Has(newLabel) {
+				gc.CreateComment(owner, repo, number, fmt.Sprintf(nonExistentLabel, newLabel))
```

Don't ignore the error. Group non-existent labels and make one comment at the end.